google/stackdriver_monitoring/spectator_client.py

```diff
@@ -19,6 +19,38 @@
 import urllib2
 
 
+def __foreach_metric_tag_binding(
+    service, metric_name, metric_data,
+    visitor, visitor_pos_args, visitor_kwargs):
+  for metric_instance in metric_data['values']:
+    visitor(service, metric_name, metric_data, metric_instance,
+            *visitor_pos_args, **visitor_kwargs)
+
+
+def foreach_metric_in_service_map(
+    service_map, visitor, *visitor_pos_args, **visitor_kwargs):
+  for service, service_metrics in service_map.items():
+    for metric_name, metric_data in service_metrics.items():
+      __foreach_metric_tag_binding(
+          service, metric_name, metric_data,
+          visitor, visitor_pos_args, visitor_kwargs)
+
+
+def normalize_name_and_tags(name, metric_metadata, metric_instance):
+  tags = metric_instance.get('tags', None)
+  if not tags:
+    return name, None   # signal this metric had no tags so we can ignore it.
+
+  is_timer = metric_metadata['kind'] == 'Timer'
+  if is_timer:
+    for index, tag in enumerate(tags):
+      if tag['key'] == 'statistic':
+        name = name + '__{0}'.format(tag['value'])
```

Comment:
Arbitrary. It is what is in kork. I wanted to convert existing names into something new and wanted to avoid collisions with other metrics since it is a blind search/replace.

In spinnaker, the Timer metrics have a tag called "statistic" with is either "count" or "totalTime" where count is the number of calls and totalTime is the nanosecond duration.
So you might have a timer metric "invocations" with values
   name='invocations', tags=[{'statistic': 'count'}] value={'time':T, 'value':C}
   name='invocations', tags=[{'statistic': 'totalTime'}] value={'time':T, 'value':N}

This feels particularly awkward. Especially because count and totalTime are different things with different units and semantics so should be treated in different ways. So I split them into two different metrics

name='invocations__count', tags=[], value={'time':T, 'value':C}
name='invocations__totalTime', tags=[], value={'time':T, 'value':N}

Note that here there are no tags. Thats why above I explicitly use 'None' as a return value indicating there were never any tags as opposed to [] which is derived to no longer have any.